src/optimizer/OptRule.h

```diff
@@ -59,12 +59,12 @@ class OptRule {
         std::vector<OptGroupNode *> newGroupNodes;
     };
 
-    StatusOr<MatchedResult> match(OptContext *ctx, const OptGroupNode *groupNode) const;
+    StatusOr<MatchedResult> match(OptContext *ctx, const OptGroupNode *groupNode);
```

Comment:
Why do you need to change these function signatures ? One thing which you should know is that all opt rules should be stateless since they are shared by all queries.

Reply:
How about removing its state after `OptRule::transfrom`?

Reply:
I missed considering this.  It will really cause some runtime error. I will reimplement it.

Reply:
Already reverted.